pgctl/service.py

```diff
@@ -150,16 +165,36 @@ def ensure_exists(self):
         if not self.path.check(dir=True):
             raise NoSuchService("No such service: '%s'" % bestrelpath(str(self.path)))
 
-        # ensure symlink {service_dir}/supervise -> {scratch_dir}/supervise
-        # this will re-connect the service to its state descriptors if the symlinks have been deleted or moved
-        supervise_in_scratch = self.scratch_dir.join('supervise')
-        supervise_in_scratch.ensure_dir()
-        symlink_if_necessary(supervise_in_scratch, self.path.join('supervise'))
+        self._ensure_supervise_is_scratch('supervise')
         self.__exists = True
 
     def ensure_logs(self):
         self.ensure_exists()
-        self.path.ensure('log')
+        self.path.ensure_dir('logs')
+        self.path.join('logs').ensure('current')
```

Comment:
So we could have logs in the log directory, it's just that we would then be using the same directory as an s6-log dir (which contains a lock file & status file), and as a service directory. That might be okay, but I like the idea of them being separate.